### PR TITLE
 Clarify arrow function brace requirements in documentation #34006

### LINKED ISSUE #34006: Clarify or replace "additional lines of processing"
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

### What specific section or headline is this issue about?

Description

### What information was incorrect, unhelpful, or incomplete?

"If the body has additional lines of processing" in "The braces can only be omitted if the function directly returns an expression. If the body has additional lines of processing, the braces are required — and so is the return keyword." is a little misleading.
Because e.g.:
```
(a) => (
    a = [3,2,1],
    delete a[1],
    a.reduce((x,y) => x+y)
)
```
Is completely valid expression, that consists of multiple expressions and has additional lines. Quick search for term "lines of processing" does not return anything relevant.

### What did you expect to see?

I would expect something like "If the body has to include statements". (Although, any program can be written with only expressions)

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/functions/arrow_functions`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/functions/arrow_functions/index.md
* Last commit: https://github.com/mdn/content/commit/e3faa375b0179de77a5eff00074e3d168a0a904c
* Document last modified: 2023-10-04T23:23:49.000Z

</details>

### PR DESCRIPTION
**Pull request by Ashish-CodeJourney**
### Description

Updated the documentation to accurately describe when braces are required in arrow functions. The previous explanation used the term "lines of processing," which was misleading. Replaced it with a clearer explanation about the inclusion of statements.

### Motivation

The term "lines of processing" was misleading because arrow functions can have multiple expressions on separate lines and still return a value directly. The requirement for braces is based on the inclusion of statements, not the number of lines.

### Additional Details

For more information on arrow functions, refer to the [MDN Web Docs on Arrow Functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions).

### Related Issues and Pull Requests

Fixes #34006

---

After submitting, go to the "Checks" tab of your PR for the build status.

### CODE REVIEW

File: files/en-us/web/javascript/reference/functions/arrow_functions/index.md [hunk 1]
```diff
@@ -152,7 +154,7 @@ const func2 = (x, y) => {
 
 Returning object literals using the expression body syntax `(params) => { object: literal }` does not work as expected.
 
-```js-nolint example-bad
+```js
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/functions/arrow_functions/index.md:157]: Why are you changing these?

Comment (Ashish-CodeJourney) [reply files/en-us/web/javascript/reference/functions/arrow_functions/index.md:157]: Sorry my bad, It was added by copilot

Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/functions/arrow_functions/index.md:157]: Please revert all other lines of changes and only keep the one that says "contains statements". AI is probably not going to do you good here.

Comment (Ashish-CodeJourney) [reply files/en-us/web/javascript/reference/functions/arrow_functions/index.md:157]: sure give me a min

File: files/en-us/web/javascript/reference/functions/arrow_functions/index.md [hunk 2]
```diff
@@ -106,7 +106,7 @@ const b = 2;
 () => a + b + 100;
 ```
 
-The braces can only be omitted if the function directly returns an expression. If the body has additional lines of processing, the braces are required — and so is the `return` keyword. Arrow functions cannot guess what or when you want to return.
+The braces can only be omitted if the function directly returns an expression. If the body contains statements, the braces are required — and so is the `return` keyword. Arrow functions cannot guess what or when you want to return.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/functions/arrow_functions/index.md:109]:
```suggestion
The braces can only be omitted if the function directly returns an expression. If the body contains anything other than a single `return` statement, the braces are required — and so is the `return` keyword. Arrow functions cannot guess what or when you want to return.
```

**Review by Josh-Cena — APPROVED** [review]
Thanks

File: files/en-us/web/javascript/reference/functions/arrow_functions/index.md [hunk 3]
```diff
@@ -106,7 +106,7 @@ const b = 2;
 () => a + b + 100;
 ```
 
-The braces can only be omitted if the function directly returns an expression. If the body has additional lines of processing, the braces are required — and so is the `return` keyword. Arrow functions cannot guess what or when you want to return.
+The braces can only be omitted if the function directly returns an expression. If the body contains anything other than a single `return` statement, the braces are required — and so is the `return` keyword. Arrow functions cannot guess what or when you want to return.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/functions/arrow_functions/index.md:109]:
```suggestion
The braces can only be omitted if the function directly returns an expression. If the body has statements, the braces are required — and so is the `return` keyword. Arrow functions cannot guess what or when you want to return.
```

**Pull request merged**